### ISSUE TITLE
Phylogenetic confounding and data leakage

### ISSUE THREAD

**Issue by ArtPoon** [PoonLab/surfaces#129]
> Phylogenetic Confounding and Data Leakage. The supervised learning analysis (k-Nearest Neighbors) claims to distinguish surface proteins of enveloped versus non-enveloped viruses, but it is heavily confounded by viral taxonomy. The "non-enveloped" dataset is overwhelmingly dominated by a single viral family (Picornaviridae), with every other virus coming from its own family. The "enveloped" group is similarly dominated by Flaviviridae. By using standard Leave-One-Out Cross-Validation (LOOCV), the authors introduce significant possible data "leakage". When predicting the envelope status of a Dengue virus protein, the model can draw upon the relatively closely related Zika and West Nile proteins in the training set. The classifier could simply be capturing family-specific phylogenetic baseline traits rather than a generalizable "envelope" signature. A strict Leave-One-Family-Out cross-validation would be much more robust.

This is a fair point.  For k-nearest neighbours, I think the correction would be to draw nearest neighbours from proteins that are not from the same virus family.  However, there may not be much data left if we exclude observations at this level.

**Comment by ArtPoon** [comment]
If the signal is being driven by virus family, then we should obtain similar results for non-exposed proteins (enveloped versus non-enveloped, not just polymerases).  If not, then that supports the conclusion that there is something different about surface-exposed proteins.

**Comment by ArtPoon** [comment]
We could also look more closely at where virus species / families are distributed in the MDS.  This is basically #146

**Comment by ArtPoon** [comment]
Actually, this is an interesting issue.  **Why is there phylogenetic confounding for evolutionary fingerprints at all?**

Phylogenetic confounding is the lack of independence among observations of an evolving characteristic, such as a nucleotide sequence or phenotype like immune escape, due to common ancestry.  

Is an evolutionary fingerprint an evolving characteristic?  It is supposed to be a representation of the selective regime that has shaped the genetic variation of a protein-coding sequence.  Completely unrelated (non-homologous) sets of sequences can have similar fingerprints.  If these alignments represent different viruses and genes, then this cannot be due to phylogenetic confounding, can it?

**Comment by hugocastelan** [comment]
The kNN classifier was evaluated using a Leave-One-Family-Out (LOFO) cross-validation strategy, in which all sequences from a given viral family (e.g., Flaviviridae or Picornaviridae) were excluded from the training set and used exclusively for testing. This approach was designed to minimize the influence of phylogenetic similarity and assess the model’s ability to generalize across evolutionary lineages. The resulting confusion matrices and performance metrics show a reduction in accuracy compared to standard cross-validation, indicating that part of the predictive signal is driven by family-level structure.   

<img width="1685" height="1344" alt="Image" src="https://github.com/user-attachments/assets/aeaaaa10-3a1c-4e15-b03d-8a5e7f46f4a6" />

**Comment by hugocastelan** [comment]
- LOOCV: In this method, only one sample is excluded, but in this case, the training data still contains very similar sequences, sometimes from the same family, that are excluded, so it is not a very rigorous method.

- LOFO: This is a much more rigorous method, where entire families are excluded, so it represents a much more difficult test for the model.

- In LOOCV, high performance is achieved, with an accuracy close to 0.83, but in LOFO, there is a significant drop in performance, especially in F1. 

<img width="1685" height="1344" alt="Image" src="https://github.com/user-attachments/assets/2e038bb6-aca1-4649-a63a-51c2d03628bc" />

**Comment by ArtPoon** [comment]
> The kNN classifier was evaluated using a Leave-One-Family-Out (LOFO) cross-validation strategy, in which all sequences from a given viral family (e.g., Flaviviridae or Picornaviridae) were excluded from the training set and used exclusively for testing. This approach was designed to minimize the influence of phylogenetic similarity and assess the model’s ability to generalize across evolutionary lineages. The resulting confusion matrices and performance metrics show a reduction in accuracy compared to standard cross-validation, indicating that part of the predictive signal is driven by family-level structure.
> 
> <img alt="Image" width="1685" height="1344" src="https://private-user-images.githubusercontent.com/64658071/571544929-aeaaaa10-3a1c-4e15-b03d-8a5e7f46f4a6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzQ5OTgzMTIsIm5iZiI6MTc3NDk5ODAxMiwicGF0aCI6Ii82NDY1ODA3MS81NzE1NDQ5MjktYWVhYWFhMTAtM2ExYy00ZTE1LWIwM2QtOGE1ZTdmNDZmNGE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzMxVDIzMDAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlmNDBjMTI1MGI0MmU2YzUyMGFkOWMwZjE5MTNhN2Y3Yjg4NGNlNDYyYWEwOTQ2NjU0MGNlYzllZWViNjE1OWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nbmAJxubQviL795DASJJxel24j6BX5Nya9ZI5WT9MHg">

What is it that you are trying to predict here?  Whether the virus is enveloped or not?  (i.e., y-axis, "TRUE")

**Comment by ArtPoon** [comment]
@hugocastelan - I think a more direct test would be to look at whether exposed and non-exposed proteins are significantly separated within Picornaviridae

**Comment by hugocastelan** [comment]
The results permanova with Picornaviridae 

``` 
Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 9999

adonis2(formula = D ~ Exposure, data = mds_picorna, permutations = 9999)
         Df SumOfSqs      R2      F Pr(>F)   
Model     1   15.569 0.15025 7.6033  0.003 **
Residual 43   88.048 0.84975                 
Total    44  103.617 1.00000                 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
``` 
<img width="1687" height="1344" alt="Image" src="https://github.com/user-attachments/assets/572b4ad3-6cdf-44d4-9f3d-41a5e41b0da9" />
